### ISSUE TITLE
Pod install failing due to undefined 'exists' method at RNGestureHandler.podspec file

### ISSUE THREAD

**Issue by ArunGovil** [software-mansion/react-native-gesture-handler#2367]
### Description

Pod install keeps failing due to an undefined  'exists' method at RNGestureHandler.podspec file. 

However the app works fine on Android, facing this error whenever I'm trying to run on ios environment.

I have checked with two of my other apps, where also I'm getting the same issue.

Here's the error log when running pod install:
```sh

[!] Invalid `Podfile` file:
[!] Invalid `RNGestureHandler.podspec` file: undefined method `exists?' for File:Class.

 #  from /Users/user/Development/QuickMark/node_modules/react-native-gesture-handler/RNGestureHandler.podspec:5
 #  -------------------------------------------
 #
 >  isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 #  if isUserApp
 #  -------------------------------------------
.

 #  from /Users/user/Development/QuickMark/ios/Podfile:8
 #  -------------------------------------------
 #  target 'quickmark' do
 >    config = use_native_modules!
 #
 #  -------------------------------------------

```

### Steps to reproduce

1. cd ios
2. pod install

### Snack or a link to a repository

https://github.com/ArunGovil/QuickMark

### Gesture Handler version

2.8.0

### React Native version

0.70.6

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

Debug mode

### Device

iOS simulator

### Device model

iPhone 14

### Acknowledgements

Yes

**Comment by official-ravi** [comment]
@ArunGovil have you got any solution for this issue? I am also facing the same issue after upgrading to latest MAC OS

**Comment by Acetyld** [comment]
Same problem here, seems like a unkown issue so far. Anyone got a temp fix? (also running newest ventura)

**Comment by lotfi-a** [comment]
same problem here :(

**Comment by devoutDisciple** [comment]
Same problem here

**Comment by aqweider** [comment]
We have same issue here

**Comment by beebop1032** [comment]
Same here

**Comment by j-piasecki** [comment]
Before #2368 gets released, this patch should do the trick:

```diff
diff --git a/RNGestureHandler.podspec b/RNGestureHandler.podspec
index fed3e8aea..fc3226de1 100644
--- a/RNGestureHandler.podspec
+++ b/RNGestureHandler.podspec
@@ -2,7 +2,7 @@ require "json"
 
 fabric_enabled = ENV['RCT_NEW_ARCH_ENABLED'] == '1'
 
-isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
+isUserApp = File.exist?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 if isUserApp
   libInstances = %x[find ../../ -name "package.json" | grep "/react-native-gesture-handler/package.json" | grep -v "/.yarn/"]
   libInstancesArray = libInstances.split("\n")
```

**Comment by witcherlin** [comment]
As a workaround, if you have `cocoapods` installed via `brew`, you can patch the ruby version in the brew formula:

1. Open a formula in vscode (for example)
```bash
HOMEBREW_EDITOR=code brew edit cocoapods
```

2. Replace `ruby` with `ruby@2.7` in lines 21 and 24 and save
```ruby
19 | ...
20 |  uses_from_macos "libffi", since: :catalina
21 |  uses_from_macos "ruby@2.7", since: :catalina
22 |
23 |  on_arm do
24 |   depends_on "ruby@2.7"
25 |  end
26 | ...
```

3. Reinstall cocoapods
```bash
brew reinstall --build-from-source cocoapods
```

In my case the problem was not only in this package, and this patch "solved" the problem for all packages.

**Comment by ArunGovil** [comment]
@j-piasecki Thank you for responding, it did worked for gesture-handler.

However the issue was still coming up in other packages.

Followed the answer by @witcherlin which solved the issue completely.  

Thanks everyone, closing the issue.

**Comment by Acetyld** [comment]
> ```shell
> ```shell
> brew reinstall --build-from-source cocoapods
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> ```

For me this sadly didint work.
I keep getting 
```
Error: Failure while executing; `code /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/cocoapods.rb /opt/homebrew/Library/Taps/homebrew/homebrew-cask/Casks/cocoapods.rb` exited with 1
```

When executing the: HOMEBREW_EDITOR=code brew edit cocoapods

**Comment by Acetyld** [comment]
> As a workaround, if you have `cocoapods` installed via `brew`, you can patch the ruby version in the brew formula:
> 
> 1. Open a formula in vscode (for example)
> 
> ```shell
> HOMEBREW_EDITOR=code brew edit cocoapods
> ```
> 
> 2. Replace `ruby` with `ruby@2.7` in lines 21 and 24 and save
> 
> ```ruby
> 19 | ...
> 20 |  uses_from_macos "libffi", since: :catalina
> 21 |  uses_from_macos "ruby@2.7", since: :catalina
> 22 |
> 23 |  on_arm do
> 24 |   depends_on "ruby@2.7"
> 25 |  end
> 26 | ...
> ```
> 
> 3. Reinstall cocoapods
> 
> ```shell
> brew reinstall --build-from-source cocoapods
> ```
> 
> In my case the problem was not only in this package, and this patch "solved" the problem for all packages.

For me this sadly didint work.
I keep getting 
```
Error: Failure while executing; `code /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/cocoapods.rb /opt/homebrew/Library/Taps/homebrew/homebrew-cask/Casks/cocoapods.rb` exited with 1
```

When executing the: HOMEBREW_EDITOR=code brew edit cocoapods


> As a workaround, if you have `cocoapods` installed via `brew`, you can patch the ruby version in the brew formula:
> 
> 1. Open a formula in vscode (for example)
> 
> ```shell
> HOMEBREW_EDITOR=code brew edit cocoapods
> ```
> 
> 2. Replace `ruby` with `ruby@2.7` in lines 21 and 24 and save
> 
> ```ruby
> 19 | ...
> 20 |  uses_from_macos "libffi", since: :catalina
> 21 |  uses_from_macos "ruby@2.7", since: :catalina
> 22 |
> 23 |  on_arm do
> 24 |   depends_on "ruby@2.7"
> 25 |  end
> 26 | ...
> ```
> 
> 3. Reinstall cocoapods
> 
> ```shell
> brew reinstall --build-from-source cocoapods
> ```
> 
> In my case the problem was not only in this package, and this patch "solved" the problem for all packages.

For me this sadly didint work.
I keep getting 
```
Error: Failure while executing; `code /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/cocoapods.rb /opt/homebrew/Library/Taps/homebrew/homebrew-cask/Casks/cocoapods.rb` exited with 1
```

When executing the: HOMEBREW_EDITOR=code brew edit cocoapods

**Comment by KiwiKilian** [comment]
@witcherlin `ruby@3.1` also works, the problem only occurs with the latest `ruby@3.2`.

**Comment by dcsan** [comment]
anyone know why the latest ruby is even used? i thought the pods installer was using system ruby?

**Comment by iway1** [comment]
better question is why ruby releases breaking changes in minor versions

**Comment by elmargasimov** [comment]
I solved it by installing a version manager for Ruby (`rbenv` in my case, but you can use `rvm` too). I had `cocoapods` installed with `brew` previously, I removed this by running `brew uninstall cocoapods`. Then installed ruby version `3.1.0` with `rbenv` (`rbenv install 3.1.0`). Then ran `gem install cocoapods`. Works fine now.

**Comment by yolpsoftware** [comment]
Uhm, thanks for the workaround and fix, but why does a deprecated method break the whole thing? Seems to me the point of a "deprecation" is exactly to warn devs that this method might be removed in a later version, so they can adapt before the pipelines of all the non-Ruby people are broken 🤷🏻‍♂️

**Comment by iway1** [comment]
ruby likes to release breaking changes in minor versions, not sure why they don't respect semver.

**Comment by MahmoudAziz1996** [comment]
Remove `^` from package dot json for react native gesture handler

Then:

```
rm -rf node_modules
yarn install 
cd ios
pod install 
```

And after that your pod will work fine.

It woks fine for me :)

**Comment by rikur** [comment]
Fixed in latest release of `react-native-gesture-handler`.

```
"resolutions": {
    "react-native-gesture-handler": "2.14.1"
  },
 ```

**Comment by erTongWang** [comment]
> Before #2368 gets released, this patch should do the trick:
> 
> ```diff
> diff --git a/RNGestureHandler.podspec b/RNGestureHandler.podspec
> index fed3e8aea..fc3226de1 100644
> --- a/RNGestureHandler.podspec
> +++ b/RNGestureHandler.podspec
> @@ -2,7 +2,7 @@ require "json"
>  
>  fabric_enabled = ENV['RCT_NEW_ARCH_ENABLED'] == '1'
>  
> -isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
> +isUserApp = File.exist?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
>  if isUserApp
>    libInstances = %x[find ../../ -name "package.json" | grep "/react-native-gesture-handler/package.json" | grep -v "/.yarn/"]
>    libInstancesArray = libInstances.split("\n")
> ```

This is a viable approach

**Comment by j-piasecki** [comment]
FYI: https://github.com/software-mansion/react-native-gesture-handler/pull/2368 was released in 2.9.0
